chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/UserDefinedTriggersEmailJob.java

```diff
@@ -0,0 +1,251 @@
+        JobDataMap jobDataMap = userTrigger.getJob().getJobDataMap();
+        if (jobDataMap != null && jobDataMap.getKeys() != null && jobDataMap.getKeys().length > 0) {
+            triggerCsv.add(Stream.of(jobDataMap.getKeys())
+                .filter(key -> key != null && jobDataMap.get(key) != null)
+                .filter(key -> !Arrays.stream(standardJobDataKeys).anyMatch(key::equalsIgnoreCase))
+                .map(key -> key + ":" + jobDataMap.get(key).toString())
+                .collect(Collectors.joining(";")));
+        } else {
+            triggerCsv.add("");
+        }
+        return triggerCsv;
+    }
+
+    private String getAcbs(ChplRepeatableTrigger trigger) {
+        if (trigger == null || trigger.getAcb() == null) {
+            return "";
+        }
+        //expect a comma-separated list of ACB IDs
+        String acbs = trigger.getAcb();
+        String[] acbList = acbs.split(",");
+        if (acbList == null || acbList.length == 0) {
+            return "";
+        }
+        return Stream.of(acbList)
+            .map(acbIdStr -> new Long(acbIdStr))
+            .map(acbId -> {
+                try {
+                    return acbDao.getById(acbId);
+                } catch (EntityRetrievalException e) {
+                    LOGGER.error("Could not find ACB with ID " + acbId, e);
+                }
+                return null;
+            })
+            .filter(acb -> acb != null)
+            .map(acb -> acb.getName())
+            .collect(Collectors.joining(";"));
+    }
+
+    private String getCronDescription(String quartzCronString) {
+        if (StringUtils.isEmpty(quartzCronString)) {
+            return "";
+        }
+
+        //get a predefined instance
+        CronDefinition cronDefinition = CronDefinitionBuilder.instanceDefinitionFor(CronType.QUARTZ);
+
+        //create a parser based on provided definition
+        CronParser parser = new CronParser(cronDefinition);
+        Cron quartzCron = parser.parse(quartzCronString);
+
+        //create a descriptor for a specific Locale
+        CronDescriptor descriptor = CronDescriptor.instance(Locale.US);
+
+        //parse some expression and ask descriptor for description
+        String description = descriptor.describe(quartzCron);
+        //description will be: "every 45 seconds"
+        return description;
+    }
+
+    private List<String> getHeaderRow() {
+        List<String> row = new ArrayList<String>();
+        row.add("Job Name");
+        row.add("Email Address");
+        row.add("ACB(s)");
+        row.add("Quartz Cron Schedule");
+        row.add("Cron Schedule Description");
+        row.add("Extra Job Data");
+        return row;
+    }
+
+    private List<String> getFooterRow() {
+        List<String> row = new ArrayList<String>();
+        row.add("");
+        row.add("");
+        row.add("");
+        row.add(env.getProperty("userTriggersReport.cronDescriptionUrl"));
```

Comment:
Does it make more sense for this to be in the email body? The csv is supposed to be the data that's in the system, and CSVs don't usually have footer rows. I feel like this kind of row will end up causing confusion with sorting / filtering. Maybe putting this URL in the body of the email with something like

> See <insert url here> for more information about what the cron schedules in the attached CSV mean